src/Symfony/Component/DependencyInjection/Argument/ServiceLocatorArgument.php

```diff
@@ -39,7 +39,7 @@ public function getValues()
     public function setValues(array $values)
     {
         foreach ($values as $v) {
-            if (!$v instanceof Reference) {
+            if (!$v instanceof Reference && null !== $v) {
```

Comment:
`null`s will create services that have "has()" return true, and "get()" return null. Not sure it's useful at all. What about skipping them instead?